unity_buy_android/unity_buy_plugin/src/main/java/com/shopify/unity/buy/AndroidPayCheckoutSession.java

```diff
@@ -0,0 +1,102 @@
+            String merchantName,
+            String summaryItemsString,
+            String currencyCode,
+            String countryCode,
+            boolean requiresShipping) {
+
+        SummaryItems items = null;
+        try {
+            items = SummaryItems.fromJsonString(summaryItemsString);
+        } catch (JSONException e) {
+            Log.e("ShopifyBuyPlugin", "Failed to parse summary items from Unity");
+            e.printStackTrace();
+            return false;
+        } catch (IllegalArgumentException e) {
+            Log.e("ShopifyBuyPlugin", e.getMessage());
+            e.printStackTrace();
+            return false;
+        }
+
+        PayCart cart = PayCart.builder()
+                .merchantName(merchantName)
+                .currencyCode(currencyCode)
+                .shippingAddressRequired(requiresShipping)
+                .subtotal(items.subtotal)
+                .shippingPrice(items.shippingPrice)
+                .taxPrice(items.taxPrice)
+                .totalPrice(items.totalPrice)
+                .build();
+
+        addAndroidPayFragment(unityDelegateObjectName, cart);
+        return true;
+    }
+
+    private void addAndroidPayFragment(String unityDelegateObjectName, PayCart cart) {
+        UnityAndroidPayFragment payFragment =
+                UnityAndroidPayFragment.newInstance(unityDelegateObjectName, cart);
+        UnityPlayer.currentActivity.getFragmentManager()
+                .beginTransaction()
+                .add(payFragment, "payFragment")
+                .commit();
+    }
+
+    // Wrapper around SummaryItems from Unity that performs validation.
+    private static class SummaryItems {
+        private BigDecimal subtotal;
+        private BigDecimal shippingPrice;
+        private BigDecimal taxPrice;
+        private BigDecimal totalPrice;
+
+        static SummaryItems fromJsonString(String jsonString) throws JSONException,
+                IllegalArgumentException  {
+            SummaryItems items = new SummaryItems();
+            JSONArray jsonArray = new JSONArray(jsonString);
+            for (int i = 0; i < jsonArray.length(); i++) {
+                JSONObject summaryItemObj = new JSONObject(jsonArray.getString(i));
+                String label = summaryItemObj.getString("Label");
+
+                if (label.equals("SUBTOTAL")) {
+                    items.subtotal = amountFromJson(summaryItemObj);
+                } else if (label.equals("SHIPPING")) {
+                    items.shippingPrice = amountFromJson(summaryItemObj);
+                } else if (label.equals("TAXES")) {
+                    items.taxPrice = amountFromJson(summaryItemObj);
+                } else if (label.equals("TOTAL")) {
+                    items.totalPrice = amountFromJson(summaryItemObj);
+                }
+            }
+
+            // Verify that we've set all of the required parameters that we need.
+            if (items.subtotal == null) {
```

Comment:
I added this validation to catch any developer errors when trying to pass missing information. Does this pattern make sense in Java?